fix(linter/plugins): output errors thrown in JS plugins

overlookmotel opened the pull request:
Previously any errors on JS side in JS plugins were silently ignored. Instead create a diagnostic including stack trace.

overlookmotel commented on crates/oxc_linter/src/lib.rs:
```diff
@@ -466,8 +466,11 @@ impl Linter {
-            Err(_err) => {
-                // TODO: report diagnostic
+            Err(err) => {
+                ctx_host.push_diagnostic(Message::new(
+                    OxcDiagnostic::error(format!("Error running JS plugin:\n{err}")),
+                    PossibleFixes::None,
+                ));
```
@camc314 Can I just check this is correct? Or can an `OxcDiagnostic::error` get filtered out in some cases?

overlookmotel commented on crates/oxc_linter/src/lib.rs:
```diff
@@ -466,8 +468,13 @@ impl Linter {
             }
-            Err(_err) => {
-                // TODO: report diagnostic
+            Err(err) => {
+                let path = path.to_string_lossy();
+                let message = format!("Error running JS plugin.\nFile path: {path}\n{err}");
+                ctx_host.push_diagnostic(Message::new(
+                    OxcDiagnostic::error(message),
+                    PossibleFixes::None,
+                ));
```
@camc314 Can I just check this is correct? Or can an `OxcDiagnostic::error` get filtered out in some cases?

camc314 commented:
### Merge activity

* **Sep 25, 8:17 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
fix(linter/plugins): output errors thrown in JS plugins (#14096)

Previously any errors on JS side in JS plugins were silently ignored. Instead create a diagnostic including stack trace.

overlookmotel commented on crates/oxc_linter/src/lib.rs:
```diff
@@ -466,8 +468,13 @@ impl Linter {
             }
-            Err(_err) => {
-                // TODO: report diagnostic
+            Err(err) => {
+                let path = path.to_string_lossy();
+                let message = format!("Error running JS plugin.\nFile path: {path}\n{err}");
+                ctx_host.push_diagnostic(Message::new(
+                    OxcDiagnostic::error(message),
+                    PossibleFixes::None,
+                ));
```
It feels odd that need to manually include path of file being linted in the error message.

I added `.with_label(SPAN)` and then it *did* include file path in output automatically, but it also then prints a code block showing the error as being on 1st line of the file, which is misleading - the error may have occurred at any point in AST visitation.

Is there a better way to do this?
